src/pokedex_screen.c

```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
+        gSprites[gUnknown_203ACF0->field_4A[14]].oam.affineMode = 1;
+        gSprites[gUnknown_203ACF0->field_4A[14]].oam.matrixNum = 2;
+        gSprites[gUnknown_203ACF0->field_4A[14]].oam.priority = 1;
+        gSprites[gUnknown_203ACF0->field_4A[14]].pos2.y = gPokedexEntries[speciesId].pokemonOffset;
+        SetOamMatrix(2, gPokedexEntries[speciesId].pokemonScale, 0, 0, gPokedexEntries[speciesId].pokemonScale);
+        gUnknown_203ACF0->field_4A[15] = CreateTrainerPicSprite(PlayerGenderToFrontTrainerPicId_Debug(gSaveBlock2Ptr->playerGender, 1), 1, 80, 104, 0, 65535);
+        gSprites[gUnknown_203ACF0->field_4A[15]].oam.paletteNum = 2;
+        gSprites[gUnknown_203ACF0->field_4A[15]].oam.affineMode = 1;
+        gSprites[gUnknown_203ACF0->field_4A[15]].oam.matrixNum = 1;
+        gSprites[gUnknown_203ACF0->field_4A[15]].oam.priority = 1;
+        gSprites[gUnknown_203ACF0->field_4A[15]].pos2.y = gPokedexEntries[speciesId].trainerOffset;
+        SetOamMatrix(1, gPokedexEntries[speciesId].trainerScale, 0, 0, gPokedexEntries[speciesId].trainerScale);
+    }
+    else
+    {
+        gUnknown_203ACF0->field_4A[14] = 0xff;
+        gUnknown_203ACF0->field_4A[15] = 0xff;
+    }
+
+    gUnknown_203ACF0->field_02[2] = sub_8134230(species, 2001, 3, v8 * 8);
+    if (!(sub_81344E0(gUnknown_203ACF0->field_02[2])))
+    {
+        BlitBitmapRectToWindow(gUnknown_203ACF0->field_4A[0], (void *)gUnknown_8443D00, 0, 0, 88, 16, 4, 28, 88, 16);
+        {
+            s32 width = GetStringWidth(0, gText_AreaUnknown, 0);
+            sub_81047C8(gUnknown_203ACF0->field_4A[0], 0, gText_AreaUnknown, (96 - width) / 2, 29, 0);
+        }
+    }
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[0], 2);
+    FillWindowPixelBuffer(1, 255);
+    sub_81047C8(1, 0, gText_Cry, 8, 2, 4);
+    sub_8104C2C(gText_CancelPreviousData);
+    PutWindowTilemap(1);
+    CopyWindowToVram(1, 2);
+
+    return 1;
+}
+
+
+u8 sub_81067C0(void)
+{
+    int i;
+
+    sub_81343F4(gUnknown_203ACF0->field_02[2]);
+
+    for (i = 0; i < 13; i++)
+        sub_81047B0(&gUnknown_203ACF0->field_4A[i]);
+    if (gUnknown_203ACF0->field_4A[15] != 0xff)
+        FreeAndDestroyTrainerPicSprite(gUnknown_203ACF0->field_4A[15]);
+    if (gUnknown_203ACF0->field_4A[14] != 0xff)
+        FreeAndDestroyMonPicSprite(gUnknown_203ACF0->field_4A[14]);
+    return 0;
+}
+
+int sub_8106810(u16 species)
+{
+    if (IsNationalPokedexEnabled() == TRUE)
+        return TRUE;
+    if (SpeciesToNationalPokedexNum(species) <= KANTO_DEX_COUNT)
+        return TRUE;
+    return FALSE;
+}
+
+u8 sub_8106838(u8 categoryNum, u8 pageNum)
+{
+    int i, count;
+    u16 species;
+
+    count = gDexCategories[categoryNum].page[pageNum].count;
+
+    for (i = 0; i < 4; i++)
+    {
+        if (i < count)
+        {
+            species = gDexCategories[categoryNum].page[pageNum].species[i];
+            if (sub_8106810(species) == TRUE)
+                if (sub_8104AB0(species, 0, 1))
+                    return 1;
```

Comment:
`if (sub_8106810(species) == TRUE && sub_8104AB0(species, 0, 1))`? Same in other uses of `sub_8106810`

Reply:
Done